crates/workspace/src/workspace.rs

```diff
@@ -2655,6 +2656,20 @@ impl Workspace {
         }))
     }
 
+    pub fn join_pane_into_next(&mut self, pane: View<Pane>, cx: &mut ViewContext<Self>) {
+        let next_pane = self
+            .find_pane_in_direction(SplitDirection::Right, cx)
+            .or_else(|| self.find_pane_in_direction(SplitDirection::Left, cx));
```

Comment:
Could we actually look for all split directions here?
I found this feature useful for closing void panes (to get one, drag and drop some unsupported file, e.g. *.ttf onto Zed)

![image](https://github.com/zed-industries/zed/assets/2690773/8915da3d-6b4c-470c-ba75-409cd90cb8ab)

but the current code cannot close certain combinations of these, e.g.

![image](https://github.com/zed-industries/zed/assets/2690773/13ff1f9f-b9cd-4f88-8bbb-8ae96658c30b)